types/node/fs/promises.d.ts

```diff
@@ -1239,6 +1242,21 @@ declare module "fs/promises" {
      * @return Fulfills with `undefined` upon success.
      */
     function cp(source: string | URL, destination: string | URL, opts?: CopyOptions): Promise<void>;
+    /**
+     * Retrieves the files matching the specified pattern.
+     */
+    function glob(pattern: string | string[]): AsyncIterableIterator<string>;
+    /**
+     * Retrieves the files matching the specified pattern.
+     */
+    function glob<Options extends GlobOptions>(
+        pattern: string | string[],
+        opt: Options,
+    ): Options extends GlobOptionsWithFileTypes
+        ? AsyncIterableIterator<Dirent>
+        : Options extends GlobOptionsWithoutFileTypes
+            ?  AsyncIterableIterator<string>
+            : AsyncIterableIterator<Dirent | string>;
```

Comment:
Semantically this should be `AsyncIterableIterator<Dirent> | AsyncIterableIterator<string>`, even though in practice it's impossible to tell the difference without consuming the iterator.